metastore/src/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -1518,6 +1518,15 @@ void addDynamicPartitions(long txnId, String dbName, String tableName, List<Stri
   void insertTable(Table table, boolean overwrite) throws MetaException;
 
   /**
+   * Performs the pre insert operation before the job starts
+   * @param table table name
+   * @param overwrite true if the insert is overwrite
+   *
+   * @throws MetaException
+   */
+  void preInsertTable(Table table, boolean overwrite) throws MetaException;
```

Comment:
I think this method doesn't belong in this interface. Since, this is not called when we are making metadata calls, instead this is tied to query lifecycle.

I think this should be added to StorageHandler interface and then in DDLTask execute(), should get called as table.getStorageHandler().preInsert()